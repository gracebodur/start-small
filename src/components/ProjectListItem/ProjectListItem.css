.ProjectListItem {
    display: flex;
    flex-direction: column;
    font-size: smaller;
    text-decoration: none;
    margin-bottom: 40px;
    margin-right: 20px;
    font-family: 'Just Another Hand', cursive;
    border: 3px solid #333333;
    border-radius: 2% 6% 5% 4% / 1% 1% 2% 4%;
    text-transform: uppercase;
    letter-spacing: 0.2ch;
    background: var(--clr-light);
    position: relative;
  }

  .ProjectListItem::before {
    content: '';
    border: 2px solid #353535;
    display: block;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate3d(-50%, -50%, 0) scale(1.015) rotate(0.5deg);
    border-radius: 1% 1% 2% 4% / 2% 6% 5% 4%;
}
  
  .ProjectListItem,
  .ProjectListItem:visited {
    color: #222831;
  }
  
  .ProjectListItem__image {
    height: 250px;
    width: 275px;
    border-radius: 10px 10px 0 0;
    background-size: cover;
    background-position: 50% 50%;
  }
  
  .ProjectListItem__details {
    padding: 10px;
    display: flex;
    flex: 1;
    flex-direction: column;
    justify-content: space-between;
    width: 275px;
  }
  
  .ProjectListItem__details h2 {
    margin: 0;
  }

  .ProjectListItem__description {
    text-transform: uppercase;
    text-align: justify;
    font-size: 1rem;
    font-weight:  var(--fw-bold);
    letter-spacing: 0.3ch;
  }
  
  .ProjectListItem__details .ProjectStarRating {
    padding: 5px 0;
    text-align: center;
    opacity: 0.85;
  }
  
  .ProjectListItem__details span {
    display: block;
    text-align: center;
    font-weight: var(--fw-bold);
    color: var(--clr-red);
  }

  @media screen 
  and (min-device-width: 1200px) 
  and (max-device-width: 1600px) 
  and (-webkit-min-device-pixel-ratio: 2)
  and (min-resolution: 192dpi) { 
    .ProjectListItem {
      margin-top: 1%;
  }
}
  