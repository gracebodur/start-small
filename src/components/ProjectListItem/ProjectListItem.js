import React, { Component } from 'react'
import { Link } from 'react-router-dom'
// import { ProjectStarRating } from '../ProjectStarRating/ProjectStarRating'
import './ProjectListItem.css'

export default class ProjectListItem extends Component { 
  render() {
    const { project } = this.props
    // console.log(this.props)

    return (
      <Link to={`/projects/${project.project_id}`} className='ProjectListItem'>
        <div className='ProjectListItem__image' style={{backgroundImage: `url(${project.imageurl})`}} />

        <div className='ProjectListItem__details'>
          <div className='ProjectListItem__text'>
            <h3 className='ProjectListItem__heading'>{project.schoolname}</h3>
            <p className='ProjectListItem__description'>{project.fulfillmenttrailer}</p>
            <p>&#8212;{project.teachername}</p>
          </div>
      
          {/* <div className='ProjectListItem__reviews'>
            <ProjectStarRating rating={project.average_review_rating} />
            <span id='ProjectListItem__review-count'>{readableReviewCount(project.number_of_reviews)}</span>
          </div> */}
        </div>
      </Link>
    )
  }
}

// function readableReviewCount(number) {
//   switch(number) {
//     case 0:
//       return 'how likely are you to donate?'

//     case 1:
//       return `how likely are you to donate?`

//     default:
//       return `based on likeliness to donate ${number}`
//   }
// }

