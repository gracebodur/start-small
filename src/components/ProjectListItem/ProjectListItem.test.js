import React from "react";
import toJson from "enzyme-to-json";
import ProjectListItem from "./ProjectListItem";

describe(`ProjectListItem component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a ProjectListItem by default", () => {
    const wrapper = <ProjectListItem />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the ProjectListItem given props", () => {
    const wrapper = <ProjectListItem {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
