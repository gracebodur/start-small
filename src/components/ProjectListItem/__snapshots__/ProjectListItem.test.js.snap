// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListItem component renders a ProjectListItem by default 1`] = `undefined`;

exports[`ProjectListItem component renders the ProjectListItem given props 1`] = `undefined`;
